src/overlay005/event_check.c

```diff
@@ -0,0 +1,1095 @@
+        sub_0205BEA8(4);
+        sub_02036BA0();
+        return TRUE;
+    }
+
+    return FALSE;
+}
+
+int FieldEventCheck_BattleTower (const FieldEventCheck *eventCheck, FieldSystem *fieldSystem)
+{
+    if (eventCheck->unused5 == FALSE && sub_0203F5C0(fieldSystem, 1) == TRUE) {
+        return TRUE;
+    }
+
+    if (eventCheck->interact) {
+        MapObject *object;
+
+        if (sub_0203CA40(fieldSystem, &object) == TRUE) {
+            if (sub_0205F588(fieldSystem->playerAvatar) == TRUE) {
+                sub_0205F5E4(fieldSystem->playerAvatar, PlayerAvatar_GetDir(fieldSystem->playerAvatar));
+            }
+            
+            if (sub_02062950(object) != 0x9) {
+                sub_0203E880(fieldSystem, sub_02062960(object), object);
+            } else {
+                sub_0203E880(fieldSystem, 0, object);
+            }
+
+            return TRUE;
+        }
+
+        int bgEvent = sub_0203CA6C(fieldSystem, (void *)sub_0203A440(fieldSystem), sub_0203A448(fieldSystem));
+        
+        if (bgEvent != 0xffff) {
+            sub_0203E880(fieldSystem, bgEvent, NULL);
+            return TRUE;
+        }
+
+        int permissionsEvent = FieldEventCheck_TilePermissionsToEvent(fieldSystem, FieldEventCheck_NextTilePermissions(fieldSystem));
+
+        if (permissionsEvent != 0xffff) {
+            sub_0203E880(fieldSystem, permissionsEvent, NULL);
+            return TRUE;
+        }
+    }
+
+    if (eventCheck->mapTransition && FieldEventCheck_MapTransition(fieldSystem, eventCheck) == TRUE) {
+        return TRUE;
+    }
+
+    if (eventCheck->registeredItem && sub_02069238(fieldSystem) == TRUE) {
+        return TRUE;
+    }
+
+    if (eventCheck->menu) {
+        Sound_PlayEffect(1533);
+        FieldMenu_Init(fieldSystem);
+        return TRUE;
+    }
+
+    return FALSE;
+}
+
+static BOOL FieldEventCheck_WildEncounter (FieldSystem *fieldSystem)
+{
+    int playerX, playerZ;
+
+    FieldEventCheck_PlayerPos(fieldSystem, &playerX, &playerZ);
+
+    if (sub_0206AE8C(SaveData_Events(fieldSystem->saveData)) == TRUE) {
+        if (sub_02056374(fieldSystem, playerX, playerZ) == TRUE) {
+            sub_02051450(fieldSystem, sub_0205639C(fieldSystem));
+            return TRUE;
+        } else {
+            return FALSE;
+        }
+    }
+
+    if (MapHeader_HasWildEncounters(fieldSystem->unk_1C->unk_00) && ov6_02240D5C(fieldSystem) == TRUE) {
+        return TRUE;
+    }
+
+    return FALSE;
+}
+
+static BOOL FieldEventCheck_MapTransition (FieldSystem *fieldSystem, const FieldEventCheck *eventCheck)
+{
+    if (eventCheck->transitionDir == DIR_NONE) {
+        return FALSE;
+    }
+
+    if (sub_02071CB4(fieldSystem, 4) == TRUE && ov8_0224BF4C(fieldSystem) == TRUE) {
+        return TRUE;
+    }
+
+    int playerX, playerZ;
+    FieldEventCheck_NextPos(fieldSystem, &playerX, &playerZ);
+
+    if (FieldSystem_CheckCollision(fieldSystem, playerX, playerZ) == FALSE) {
+        return FALSE;
+    }
+
+    u8 permissions;
```

Comment:
We've talked about this on Discord, and I do think `behavior` is a better name for this general concept. We can table that for now, though.